### DISCUSSION TITLE
Dependencies

### DISCUSSION THREAD

**Discussion by kennammp** [rohit-gohri/redocusaurus#361] [Q&A] [answered]
Hi all, I'm trying to install redocusaurus on docusaurus 2.3.1. I have react 18.3.1.

I got this error which seemed to be telling me i was running react 17.0.2 and i needed 18.0.0:
```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: alef-docs@0.0.0
npm ERR! Found: react@17.0.2
npm ERR! node_modules/react
npm ERR!   react@"^17.0.2" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^18.0.0" from @docusaurus/theme-common@3.3.2
npm ERR! node_modules/@docusaurus/theme-common
npm ERR!   peer @docusaurus/theme-common@"^3.0.0" from redocusaurus@2.0.2
npm ERR!   node_modules/redocusaurus
npm ERR!     redocusaurus@"*" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR! See C:\Users\Mattmc\AppData\Local\npm-cache\eresolve-report.txt for a full report.
```

I noticed that my package.json and package-lock.json files still had `react` and `react-dom` set to `^17.0.2`, so i changed it to `^18.3.1`

Now i'm getting this, which seems to be saying i need to be below 17.0.0:
```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: alef-docs@0.0.0
npm ERR! Found: react@18.3.1
npm ERR! node_modules/react
npm ERR!   react@"^18.3.1" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.8.4 || ^17.0.0" from @docusaurus/core@2.2.0
npm ERR! node_modules/@docusaurus/core
npm ERR!   @docusaurus/core@"2.2.0" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR! See C:\Users\Mattmc\AppData\Local\npm-cache\eresolve-report.txt for a full report.
```

what version on react do I need to be on? Is there a list of prerequisites for redocusaurus anywhere? I can't see it  here or on the redocusuarus docs site.
thanks

**Comment by rohit-gohri** [comment, marked as answer]
Hi,

The latest version of redocusaurus (v2.x) only supports docusaurus v3.x (which supports only react 18+ - https://docusaurus.io/docs/migration/v3)

If you want to use redocusaurus with docusaurus v2.x then install v1.x.

You can always check the peer dependencies to see what version is compatible with which - https://github.com/rohit-gohri/redocusaurus/blob/863fe652a5372fc3b700b4e9f3e4a3efcf10363c/packages/redocusaurus/package.json#L47-L50

**Reply by kennammp** [reply]
ok thanks Rohit
